mantidproject/mantid
# Add new `LRReductionWithReference` algorithm

**Description of work.**

Adding a new `LRReductionWithReference` algorithm outlined in Issue #28626 

**To test:**

*NOTE: Must have [`refl1d`](https://github.com/reflectometry/refl1d) package installed. Can be installed via `pip`*

We tested against instrument scientist's normalization and sample data and able to reproduce their curves at SNS's liquids reflectometer: 
![image](https://user-images.githubusercontent.com/17032543/83455371-c8f0e180-a42b-11ea-9178-8eaeb7eb2c44.png)

The above plot was produced from the following script from SNS REF_L instrument data(*NOTE*: `txt` extension is just to "trick" GitHub so we can upload the python script):
[test_lr_reduction_with_reference.py.txt](https://github.com/mantidproject/mantid/files/4717198/test_lr_reduction_with_reference.py.txt)


Try to test the input parameters for a dataset given in the system test data.
See if anything does not alter the output (change flags and make sure they fail the system test)

Fixes #28626 
---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
